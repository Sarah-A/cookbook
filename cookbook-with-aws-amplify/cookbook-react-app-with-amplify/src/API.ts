/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type CreateRecipeInput = {
  id?: string | null,
  name: string,
  rating?: number | null,
  prepTimeInMinutes?: number | null,
  cookTimeInMinutes?: number | null,
  serves?: number | null,
  description?: string | null,
  images?: Array< string | null > | null,
};

export type ModelRecipeConditionInput = {
  name?: ModelStringInput | null,
  rating?: ModelIntInput | null,
  prepTimeInMinutes?: ModelIntInput | null,
  cookTimeInMinutes?: ModelIntInput | null,
  serves?: ModelIntInput | null,
  description?: ModelStringInput | null,
  images?: ModelStringInput | null,
  and?: Array< ModelRecipeConditionInput | null > | null,
  or?: Array< ModelRecipeConditionInput | null > | null,
  not?: ModelRecipeConditionInput | null,
};

export type ModelStringInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export enum ModelAttributeTypes {
  binary = "binary",
  binarySet = "binarySet",
  bool = "bool",
  list = "list",
  map = "map",
  number = "number",
  numberSet = "numberSet",
  string = "string",
  stringSet = "stringSet",
  _null = "_null",
}


export type ModelSizeInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
};

export type ModelIntInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
};

export type Recipe = {
  __typename: "Recipe",
  id: string,
  name: string,
  rating?: number | null,
  prepTimeInMinutes?: number | null,
  cookTimeInMinutes?: number | null,
  serves?: number | null,
  description?: string | null,
  images?: Array< string | null > | null,
  createdAt: string,
  updatedAt: string,
  owner?: string | null,
};

export type UpdateRecipeInput = {
  id: string,
  name?: string | null,
  rating?: number | null,
  prepTimeInMinutes?: number | null,
  cookTimeInMinutes?: number | null,
  serves?: number | null,
  description?: string | null,
  images?: Array< string | null > | null,
};

export type DeleteRecipeInput = {
  id: string,
};

export type ModelRecipeFilterInput = {
  id?: ModelIDInput | null,
  name?: ModelStringInput | null,
  rating?: ModelIntInput | null,
  prepTimeInMinutes?: ModelIntInput | null,
  cookTimeInMinutes?: ModelIntInput | null,
  serves?: ModelIntInput | null,
  description?: ModelStringInput | null,
  images?: ModelStringInput | null,
  and?: Array< ModelRecipeFilterInput | null > | null,
  or?: Array< ModelRecipeFilterInput | null > | null,
  not?: ModelRecipeFilterInput | null,
};

export type ModelIDInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export type ModelRecipeConnection = {
  __typename: "ModelRecipeConnection",
  items?:  Array<Recipe | null > | null,
  nextToken?: string | null,
};

export type CreateRecipeMutationVariables = {
  input: CreateRecipeInput,
  condition?: ModelRecipeConditionInput | null,
};

export type CreateRecipeMutation = {
  createRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    rating?: number | null,
    prepTimeInMinutes?: number | null,
    cookTimeInMinutes?: number | null,
    serves?: number | null,
    description?: string | null,
    images?: Array< string | null > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type UpdateRecipeMutationVariables = {
  input: UpdateRecipeInput,
  condition?: ModelRecipeConditionInput | null,
};

export type UpdateRecipeMutation = {
  updateRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    rating?: number | null,
    prepTimeInMinutes?: number | null,
    cookTimeInMinutes?: number | null,
    serves?: number | null,
    description?: string | null,
    images?: Array< string | null > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type DeleteRecipeMutationVariables = {
  input: DeleteRecipeInput,
  condition?: ModelRecipeConditionInput | null,
};

export type DeleteRecipeMutation = {
  deleteRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    rating?: number | null,
    prepTimeInMinutes?: number | null,
    cookTimeInMinutes?: number | null,
    serves?: number | null,
    description?: string | null,
    images?: Array< string | null > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type GetRecipeQueryVariables = {
  id: string,
};

export type GetRecipeQuery = {
  getRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    rating?: number | null,
    prepTimeInMinutes?: number | null,
    cookTimeInMinutes?: number | null,
    serves?: number | null,
    description?: string | null,
    images?: Array< string | null > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type ListRecipesQueryVariables = {
  filter?: ModelRecipeFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListRecipesQuery = {
  listRecipes?:  {
    __typename: "ModelRecipeConnection",
    items?:  Array< {
      __typename: "Recipe",
      id: string,
      name: string,
      rating?: number | null,
      prepTimeInMinutes?: number | null,
      cookTimeInMinutes?: number | null,
      serves?: number | null,
      description?: string | null,
      images?: Array< string | null > | null,
      createdAt: string,
      updatedAt: string,
      owner?: string | null,
    } | null > | null,
    nextToken?: string | null,
  } | null,
};

export type OnCreateRecipeSubscriptionVariables = {
  owner: string,
};

export type OnCreateRecipeSubscription = {
  onCreateRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    rating?: number | null,
    prepTimeInMinutes?: number | null,
    cookTimeInMinutes?: number | null,
    serves?: number | null,
    description?: string | null,
    images?: Array< string | null > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type OnUpdateRecipeSubscriptionVariables = {
  owner: string,
};

export type OnUpdateRecipeSubscription = {
  onUpdateRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    rating?: number | null,
    prepTimeInMinutes?: number | null,
    cookTimeInMinutes?: number | null,
    serves?: number | null,
    description?: string | null,
    images?: Array< string | null > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type OnDeleteRecipeSubscriptionVariables = {
  owner: string,
};

export type OnDeleteRecipeSubscription = {
  onDeleteRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    rating?: number | null,
    prepTimeInMinutes?: number | null,
    cookTimeInMinutes?: number | null,
    serves?: number | null,
    description?: string | null,
    images?: Array< string | null > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};
