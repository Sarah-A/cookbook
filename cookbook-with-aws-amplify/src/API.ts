/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type CreateIngredientInput = {
  id?: string | null,
  name: string,
  type: IngredientType,
  description?: string | null,
  brand?: string | null,
};

export enum IngredientType {
  FRUIT_AND_VEG = "FRUIT_AND_VEG",
  MEAT = "MEAT",
  SEAFOOD = "SEAFOOD",
  BAKERY = "BAKERY",
  DAIRY = "DAIRY",
  EGGS = "EGGS",
  PANTRY = "PANTRY",
  CHILLED = "CHILLED",
  FROZEN = "FROZEN",
}


export type ModelIngredientConditionInput = {
  name?: ModelStringInput | null,
  type?: ModelIngredientTypeInput | null,
  description?: ModelStringInput | null,
  brand?: ModelStringInput | null,
  and?: Array< ModelIngredientConditionInput | null > | null,
  or?: Array< ModelIngredientConditionInput | null > | null,
  not?: ModelIngredientConditionInput | null,
};

export type ModelStringInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export enum ModelAttributeTypes {
  binary = "binary",
  binarySet = "binarySet",
  bool = "bool",
  list = "list",
  map = "map",
  number = "number",
  numberSet = "numberSet",
  string = "string",
  stringSet = "stringSet",
  _null = "_null",
}


export type ModelSizeInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
};

export type ModelIngredientTypeInput = {
  eq?: IngredientType | null,
  ne?: IngredientType | null,
};

export type Ingredient = {
  __typename: "Ingredient",
  id: string,
  name: string,
  type: IngredientType,
  description?: string | null,
  brand?: string | null,
  createdAt: string,
  updatedAt: string,
  version: number,
  owner?: string | null,
};

export type UpdateIngredientInput = {
  id: string,
  name?: string | null,
  type?: IngredientType | null,
  description?: string | null,
  brand?: string | null,
  expectedVersion: number,
};

export type DeleteIngredientInput = {
  id: string,
  expectedVersion: number,
};

export type CreateRecipeInput = {
  id?: string | null,
  name: string,
  description?: string | null,
  image?: string | null,
  serves: number,
  ingredients: Array< IngredientWithAmountInput >,
  method: string,
};

export type IngredientWithAmountInput = {
  id: string,
  amount: number,
  unit: IngredientUnitsType,
};

export enum IngredientUnitsType {
  UNITS = "UNITS",
  SPOONS = "SPOONS",
  TEA_SPOONS = "TEA_SPOONS",
  GR = "GR",
  KG = "KG",
  CAN = "CAN",
  CUP = "CUP",
}


export type ModelRecipeConditionInput = {
  name?: ModelStringInput | null,
  description?: ModelStringInput | null,
  image?: ModelStringInput | null,
  serves?: ModelIntInput | null,
  method?: ModelStringInput | null,
  and?: Array< ModelRecipeConditionInput | null > | null,
  or?: Array< ModelRecipeConditionInput | null > | null,
  not?: ModelRecipeConditionInput | null,
};

export type ModelIntInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
};

export type Recipe = {
  __typename: "Recipe",
  id: string,
  name: string,
  description?: string | null,
  image?: string | null,
  serves: number,
  ingredients:  Array<IngredientWithAmount >,
  method: string,
  createdAt: string,
  updatedAt: string,
  version: number,
  owner?: string | null,
};

export type IngredientWithAmount = {
  __typename: "IngredientWithAmount",
  id: string,
  ingredient: Ingredient,
  amount: number,
  unit: IngredientUnitsType,
};

export type UpdateRecipeInput = {
  id: string,
  name?: string | null,
  description?: string | null,
  image?: string | null,
  serves?: number | null,
  ingredients?: Array< IngredientWithAmountInput > | null,
  method?: string | null,
  expectedVersion: number,
};

export type DeleteRecipeInput = {
  id: string,
  expectedVersion: number,
};

export type ModelIngredientFilterInput = {
  id?: ModelIDInput | null,
  name?: ModelStringInput | null,
  type?: ModelIngredientTypeInput | null,
  description?: ModelStringInput | null,
  brand?: ModelStringInput | null,
  and?: Array< ModelIngredientFilterInput | null > | null,
  or?: Array< ModelIngredientFilterInput | null > | null,
  not?: ModelIngredientFilterInput | null,
};

export type ModelIDInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export type ModelIngredientConnection = {
  __typename: "ModelIngredientConnection",
  items?:  Array<Ingredient | null > | null,
  nextToken?: string | null,
};

export type ModelRecipeFilterInput = {
  id?: ModelIDInput | null,
  name?: ModelStringInput | null,
  description?: ModelStringInput | null,
  image?: ModelStringInput | null,
  serves?: ModelIntInput | null,
  method?: ModelStringInput | null,
  and?: Array< ModelRecipeFilterInput | null > | null,
  or?: Array< ModelRecipeFilterInput | null > | null,
  not?: ModelRecipeFilterInput | null,
};

export type ModelRecipeConnection = {
  __typename: "ModelRecipeConnection",
  items?:  Array<Recipe | null > | null,
  nextToken?: string | null,
};

export type CreateIngredientMutationVariables = {
  input: CreateIngredientInput,
  condition?: ModelIngredientConditionInput | null,
};

export type CreateIngredientMutation = {
  createIngredient?:  {
    __typename: "Ingredient",
    id: string,
    name: string,
    type: IngredientType,
    description?: string | null,
    brand?: string | null,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type UpdateIngredientMutationVariables = {
  input: UpdateIngredientInput,
  condition?: ModelIngredientConditionInput | null,
};

export type UpdateIngredientMutation = {
  updateIngredient?:  {
    __typename: "Ingredient",
    id: string,
    name: string,
    type: IngredientType,
    description?: string | null,
    brand?: string | null,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type DeleteIngredientMutationVariables = {
  input: DeleteIngredientInput,
  condition?: ModelIngredientConditionInput | null,
};

export type DeleteIngredientMutation = {
  deleteIngredient?:  {
    __typename: "Ingredient",
    id: string,
    name: string,
    type: IngredientType,
    description?: string | null,
    brand?: string | null,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type CreateRecipeMutationVariables = {
  input: CreateRecipeInput,
  condition?: ModelRecipeConditionInput | null,
};

export type CreateRecipeMutation = {
  createRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients:  Array< {
      __typename: "IngredientWithAmount",
      id: string,
      ingredient:  {
        __typename: "Ingredient",
        id: string,
        name: string,
        type: IngredientType,
        description?: string | null,
        brand?: string | null,
        createdAt: string,
        updatedAt: string,
        version: number,
        owner?: string | null,
      },
      amount: number,
      unit: IngredientUnitsType,
    } >,
    method: string,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type UpdateRecipeMutationVariables = {
  input: UpdateRecipeInput,
  condition?: ModelRecipeConditionInput | null,
};

export type UpdateRecipeMutation = {
  updateRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients:  Array< {
      __typename: "IngredientWithAmount",
      id: string,
      ingredient:  {
        __typename: "Ingredient",
        id: string,
        name: string,
        type: IngredientType,
        description?: string | null,
        brand?: string | null,
        createdAt: string,
        updatedAt: string,
        version: number,
        owner?: string | null,
      },
      amount: number,
      unit: IngredientUnitsType,
    } >,
    method: string,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type DeleteRecipeMutationVariables = {
  input: DeleteRecipeInput,
  condition?: ModelRecipeConditionInput | null,
};

export type DeleteRecipeMutation = {
  deleteRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients:  Array< {
      __typename: "IngredientWithAmount",
      id: string,
      ingredient:  {
        __typename: "Ingredient",
        id: string,
        name: string,
        type: IngredientType,
        description?: string | null,
        brand?: string | null,
        createdAt: string,
        updatedAt: string,
        version: number,
        owner?: string | null,
      },
      amount: number,
      unit: IngredientUnitsType,
    } >,
    method: string,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type GetIngredientQueryVariables = {
  id: string,
};

export type GetIngredientQuery = {
  getIngredient?:  {
    __typename: "Ingredient",
    id: string,
    name: string,
    type: IngredientType,
    description?: string | null,
    brand?: string | null,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type ListIngredientsQueryVariables = {
  filter?: ModelIngredientFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListIngredientsQuery = {
  listIngredients?:  {
    __typename: "ModelIngredientConnection",
    items?:  Array< {
      __typename: "Ingredient",
      id: string,
      name: string,
      type: IngredientType,
      description?: string | null,
      brand?: string | null,
      createdAt: string,
      updatedAt: string,
      version: number,
      owner?: string | null,
    } | null > | null,
    nextToken?: string | null,
  } | null,
};

export type GetRecipeQueryVariables = {
  id: string,
};

export type GetRecipeQuery = {
  getRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients:  Array< {
      __typename: "IngredientWithAmount",
      id: string,
      ingredient:  {
        __typename: "Ingredient",
        id: string,
        name: string,
        type: IngredientType,
        description?: string | null,
        brand?: string | null,
        createdAt: string,
        updatedAt: string,
        version: number,
        owner?: string | null,
      },
      amount: number,
      unit: IngredientUnitsType,
    } >,
    method: string,
    createdAt: string,
    updatedAt: string,
    version: number,
    owner?: string | null,
  } | null,
};

export type ListRecipesQueryVariables = {
  filter?: ModelRecipeFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListRecipesQuery = {
  listRecipes?:  {
    __typename: "ModelRecipeConnection",
    items?:  Array< {
      __typename: "Recipe",
      id: string,
      name: string,
      description?: string | null,
      image?: string | null,
      serves: number,
      ingredients:  Array< {
        __typename: "IngredientWithAmount",
        id: string,
        amount: number,
        unit: IngredientUnitsType,
      } >,
      method: string,
      createdAt: string,
      updatedAt: string,
      version: number,
      owner?: string | null,
    } | null > | null,
    nextToken?: string | null,
  } | null,
};
