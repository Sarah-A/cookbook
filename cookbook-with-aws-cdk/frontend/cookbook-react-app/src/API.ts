/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type NewRecipeInput = {
  id: string,
  name: string,
  description?: string | null,
  serves: number,
  ingredients: Array< string >,
  method: string,
};

export type Recipe = {
  __typename: "Recipe",
  id: string,
  name: string,
  description?: string | null,
  image?: string | null,
  serves: number,
  ingredients: Array< string >,
  method: string,
};

export type UpdateRecipeInput = {
  name?: string | null,
  description?: string | null,
  image?: string | null,
  serves?: number | null,
  ingredients?: Array< string | null > | null,
  method?: string | null,
};

export type CreateRecipeMutationVariables = {
  recipe: NewRecipeInput,
};

export type CreateRecipeMutation = {
  createRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients: Array< string >,
    method: string,
  } | null,
};

export type UpdateRecipeMutationVariables = {
  id: string,
  recipe: UpdateRecipeInput,
};

export type UpdateRecipeMutation = {
  updateRecipe?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients: Array< string >,
    method: string,
  } | null,
};

export type DeleteRecipeMutationVariables = {
  id: string,
};

export type DeleteRecipeMutation = {
  deleteRecipe?: string | null,
};

export type GetAllRecipesQuery = {
  getAllRecipes?:  Array< {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients: Array< string >,
    method: string,
  } | null > | null,
};

export type GetRecipeByIdQueryVariables = {
  id: string,
};

export type GetRecipeByIdQuery = {
  getRecipeById?:  {
    __typename: "Recipe",
    id: string,
    name: string,
    description?: string | null,
    image?: string | null,
    serves: number,
    ingredients: Array< string >,
    method: string,
  } | null,
};
