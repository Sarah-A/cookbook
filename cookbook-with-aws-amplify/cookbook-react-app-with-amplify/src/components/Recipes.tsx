import { FunctionComponent } from "react";
import { Recipe as RecipeType } from "../API";
import Recipe from "./Recipe";



const Recipes: FunctionComponent<{recipes: RecipeType[]}> = ({recipes}) => {
    return (
        <div>
            {              
                !recipes.length 
                    ? (<h2>No recipes found!</h2> ) 
                    : ( 
                        <div className="row">
                        {
                            recipes.map((recipe, index) => (
                                <Recipe 
                                    key={recipe.id}
                                    id={recipe.id}
                                    name={recipe.name}
                                    rating={recipe.rating}
                                    prepTimeInMinutes={recipe.prepTimeInMinutes}
                                    cookTimeInMinutes={recipe.cookTimeInMinutes}
                                    image={recipe.images?.length ? recipe.images[0] : null}
                                    data-testid={`recipe-${index}`}
                                />
                            ))
                        }
                        </div>
                        )
                }                 
        </div>
    );
}

export default Recipes;