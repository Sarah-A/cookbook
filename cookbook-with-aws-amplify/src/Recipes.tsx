import React, { FunctionComponent, useEffect } from 'react';
import { API, graphqlOperation } from 'aws-amplify';
import { listRecipes } from './graphql/queries';
import { Recipe } from './API';

interface IRecipes {
    data: {
        listRecipes: {
            items: Recipe;
        };
    };
};

const Recipes: FunctionComponent = () => {
    // const [recipes, setRecipes] = useState([]);

    const fetchRecipes = async () => {
        try {
            const recipesData = await API.graphql(graphqlOperation(listRecipes)) as IRecipes;
            const recipesList = recipesData.data.listRecipes.items;
            console.log("recipes list: ", recipesList);
            // setRecipes(recipesList);
        } catch (error) {
            console.log("error on getting recipes: ", error);
        }
    }
    
    useEffect(() => {
        void fetchRecipes();
    }, []);

    return (
        <div>
            <p>in Recipes!</p>
        </div>
    );
}

export default Recipes;