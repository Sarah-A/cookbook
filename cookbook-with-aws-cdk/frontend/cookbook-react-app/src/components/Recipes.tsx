import { FunctionComponent } from "react";
import API, { graphqlOperation } from '@aws-amplify/api';
import { useEffect } from 'react';

const GetAllRecipes = `query getAllRecipes {
    getAllRecipes {
      id
      name
      description
      ingredients
      method
      serves
    }
}`;


async function getAllRecipes() {
    console.log("Before calling API");
    try {
        const allRecipes = await API.graphql(graphqlOperation(GetAllRecipes));
        console.log("After calling API");
        console.log(allRecipes);    
    } catch (error) {
        console.log(error);
    }
}

const Recipes : FunctionComponent = () => {

    useEffect(() => {
        void getAllRecipes();
    }, [])

    return (
        <div className="text-warning">
            Queried Recipes!!
        </div>
    );
}

export default Recipes;