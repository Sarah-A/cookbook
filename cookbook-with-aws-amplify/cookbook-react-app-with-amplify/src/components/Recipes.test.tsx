import { render , screen } from "@testing-library/react";
import { createRenderer } from "react-test-renderer/shallow";
import Recipes from "./Recipes";
import { Recipe as RecipeType } from "../API";
// import { ReactElement } from "react";

const TEST_RECIPE_NORMAL: RecipeType = {
    __typename: "Recipe",
    id: "1",
    name: "Recipe1",
    rating: 1,
    prepTimeInMinutes: 2,
    cookTimeInMinutes: 3,
    images: ["1.jpg"],
    createdAt: "123",
    updatedAt: "123"
};

const TEST_RECIPE_WITHOUT_IMAGES_AND_RATING: RecipeType  = {
    __typename: "Recipe",
    id: "2",
    name: "Recipe2",    
    prepTimeInMinutes: 3,
    cookTimeInMinutes: 4,
    createdAt: "123",
    updatedAt: "123"
};

const TEST_RECIPE_WITH_EMPTY_IMAGES_AND_NO_COOK_TIME: RecipeType  = {
    __typename: "Recipe",
    id: "3",
    name: "Recipe3",
    rating: 3,
    prepTimeInMinutes: 4,    
    images: [],
    createdAt: "123",
    updatedAt: "123"
};
const TEST_RECIPES : RecipeType[] = [ 
    TEST_RECIPE_NORMAL, 
    TEST_RECIPE_WITHOUT_IMAGES_AND_RATING, 
    TEST_RECIPE_WITH_EMPTY_IMAGES_AND_NO_COOK_TIME
];

// const mockRecipe = {
//     __esModule: true,
//     default: jest.fn(() => {
//         <div data-testid="Recipe" />
//     })
// };
// jest.mock("./Recipe", () => () => mockRecipe);

// function verifyRecipeComponent(expectedRecipe : RecipeType , receivedElement: HTMLElement) : void {
//     expect(receivedElement.getAttribute("id")).toEqual(expectedRecipe.id);
//     if (expectedRecipe.cookTimeInMinutes !== undefined) {
//         expect(receivedElement.getAttribute("cookTimeInMinutes"))
//             .toEqual(expectedRecipe.cookTimeInMinutes?.toString());
//     } else {
//         expect(receivedElement.getAttribute("cookTimeInMinutes"))
//             .toBe("");
//     }
//     expect(receivedElement.getAttribute("id")).toEqual(expectedRecipe.id);
//     expect(receivedElement.getAttribute("id")).toEqual(expectedRecipe.id);
//     expect(receivedElement.getAttribute("id")).toEqual(expectedRecipe.id);
//     expect(receivedElement.getAttribute("id")).toEqual(expectedRecipe.id);
//     expect(receivedElement.getAttribute("id")).toEqual(expectedRecipe.id);
// }

describe("test Recipes component", () => {

    beforeEach(() => {
        jest.resetAllMocks();
    })


    test("displays 'No recipes found!` when no recipes", () => {
        const recipes = render(<Recipes recipes={[]} />);
        const notFoundMessage = recipes.getByText(/No recipes found!/i);
        expect(notFoundMessage).toBeInTheDocument();
    });

    test.skip("When available recipes, should call the expected component", () => {

        
        const Recipe = require("./Recipe");
        
        render(<Recipes recipes={TEST_RECIPES} />);

        const renderedRecipes = screen.getAllByTestId("Recipe");
        expect(renderedRecipes.length).toBe(TEST_RECIPES.length);
        expect(Recipe()).toBeCalledTimes(TEST_RECIPES.length);
        
        // for(let i=0 ; i<TEST_RECIPES.length ; i++) {
        //     const recipeElement = recipes.getByTestId("recipe-" + i.toString());
        //     const testRecipe = TEST_RECIPES.find(element => {
        //         element.id === recipeElement.getAttribute("id");
        //     })
        //     expect(testRecipe).toBeDefined();
        //     verifyRecipeComponent(testRecipe as Recipe, recipeElement);

        // }        
    });
    
    test("When available recipes, should match expected snapshot", () => {
        const renderer = createRenderer();
        renderer.render(<Recipes recipes={TEST_RECIPES} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
   
});