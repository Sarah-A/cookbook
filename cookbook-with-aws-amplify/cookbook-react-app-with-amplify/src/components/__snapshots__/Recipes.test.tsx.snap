// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Recipes component When available recipes, should match expected snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <Recipe
      cookTimeInMinutes={3}
      data-testid="recipe-0"
      id="1"
      image="1.jpg"
      name="Recipe1"
      prepTimeInMinutes={2}
      rating={1}
    />
    <Recipe
      cookTimeInMinutes={4}
      data-testid="recipe-1"
      id="2"
      image={null}
      name="Recipe2"
      prepTimeInMinutes={3}
    />
    <Recipe
      data-testid="recipe-2"
      id="3"
      image={null}
      name="Recipe3"
      prepTimeInMinutes={4}
      rating={3}
    />
  </div>
</div>
`;
